pkg/controller/dnszone/gcpactuator.go

```diff
@@ -119,24 +119,28 @@ func (a *GCPActuator) Delete() error {
 
 func (a *GCPActuator) deleteRecordSets(logger log.FieldLogger) error {
 	logger.Info("Deleting recordsets in managedzone")
+	return DeleteGCPRecordSets(a.gcpClient, a.dnsZone, logger)
+}
+
+func DeleteGCPRecordSets(gcpClient gcpclient.Client, dnsZone *hivev1.DNSZone, logger log.FieldLogger) error {
 	listOpts := gcpclient.ListResourceRecordSetsOptions{}
 	for {
-		listOutput, err := a.gcpClient.ListResourceRecordSets(a.managedZone.Name, listOpts)
+		listOutput, err := gcpClient.ListResourceRecordSets(*dnsZone.Status.GCP.ZoneName, listOpts)
```

Comment:
This change looks susceptible to the same panic experienced with the AWS changes.